packages/app/src/game/module/chat/sagas.ts

```diff
@@ -0,0 +1,27 @@
+import { take, delay, select, put, race } from "redux-saga/effects";
+import { commands as quickChatCommands } from "./state";
+import { PlayerEvents } from "@creature-chess/gamemode";
+
+
+export const handleQuickChat = function*() {
+	let action = yield take(PlayerEvents.playerReceiveQuickChatEvent.toString());
+	while (true) {
+
+		const { sendingPlayerId, chatValue } = action.payload;
+		if (!chatValue) {
+			return;
```

Comment:
You almost definitely don't want a `return` here - this will exit out of the `handleQuickChat` function entirely, whereas it needs to work as an infinite loop

You probably want to just do this, which will wait for a new action and then start the loop again (just like at the end of the saga)

```ts
if (!chatValue) {
    action = yield take(PlayerEvents.playerReceiveQuickChatEvent.toString());
    continue;
}
```